RFC-0001-prsestissimo-metrics.md

```diff
@@ -0,0 +1,355 @@
+
+Given the above estimates, we decided to keep the stats in-memory. We can revisit this if the memory footprint grows.
+
+Note: Estimating the size of a histogram with quantiles is not straight forward, it depends on rotation time and error tolerance settings.
+
+
+### 
+**Metric timestamps**
+
+Since we overwrite metrics in our current design, it is likely that Prometheus may miss updates as it pulls metrics at regular intervals. The timestamps are assigned by the Prometheus server (this is recommended) and may not reflect the exact timestamp at which the metric was recorded. On the other hand, we can maintain a list of metric values and timestamps seen so far in Prestissimo and in the response to the Prometheus we can include these &lt;timestamp, value> pairs. Prometheus must be configured to honour these timestamps. But this approach is not recommended.
+
+Prestissimo emits metric values without timestamps. Prometheus DB will be assigning timestamps to the metric values after it pulls from the Prestissimo endpoint.
+
+
+### 
+**Serialization Format**
+
+
+#### 
+**Serialize using custom interface**
+
+By default, we shall implement JSON and Prometheus Data Model serialisation of metrics. We shall expose a new Serialization interface that users can implement to customize serialization. Pros: In-house and no external dependencies. Cons: It could be challenging to keep it in sync with Prometheus data model versions. Adding support for histogram quantiles is not simple.
+
+
+#### 
+**Serializing using Prometheus-CPP: (Preferred)**
+
+Pros: Popular and simplifies histogram and summary metric maintenance. The library has implemented unit tests and integration tests for serialisation and data correctness. Cons: External dependency.
+
+Both of these approaches are prototyped in [this PR](https://github.com/prestodb/presto/pull/21599/files#).
+
+
+## 
+**Configuration**
+
+**Runtime configs:**
+1. Currently, we have `runtime-metrics-collection-enabled` configuration property in Native Presto, which when set to true, starts recording metrics.
+   
+**Compile-time configs:**
+A compile to time config PRESTO_ENABLE_PROMETHEUS, which when turned ON, includes prometheus-metrics directory and registers PrometheusReporter as the metrics reporter.
+
+**Future Work:**
+1. Mechanism to white list metrics. 
+
+
+
+## **Prototype test results**
+
+Prometheus client end visualisation of Prestissimo metrics.
+<img width="1689" alt="Screenshot 2024-01-12 at 4 44 57 PM" src="https://github.com/karteekmurthys/rfcs/assets/6971561/b58074b9-5900-49b9-9941-09c1ebdd5747">
+
+
+Current PR:[ https://github.com/prestodb/presto/pull/21599](https://github.com/prestodb/presto/pull/21599)
```

Comment:
Remove this.

Reply:
Done.

Reply:
I still see this.

Reply:
I mean't we can remove the whole line. This is not the current PR. We reference the current PR above.